haskell-wrapper/tests/MultiScalarMultiplication.hs

```diff
@@ -0,0 +1,53 @@
+module MultiScalarMultiplication (testMultiScalarMultiplication) where
+
+import           Data.Kind                                   (Type)
+import           GHC.TypeNats                                (KnownNat)
+import           Numeric.Natural                             (Natural)
+import           Prelude                                     hiding (Num (..), sum)
+import           RustFunctions                               (rustMultiScalarMultiplication)
+import           Test.Hspec                                  (describe, hspec, it, shouldReturn)
+import           Test.QuickCheck                             (Testable (property))
+
+import           ZkFold.Base.Algebra.Basic.Class             (AdditiveSemigroup ((+)), sum)
+import           ZkFold.Base.Algebra.Basic.Field             (Zp)
+import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381 (BLS12_381_Base, BLS12_381_G1, BLS12_381_Scalar)
+import           ZkFold.Base.Algebra.EllipticCurve.Class     (EllipticCurve (BaseField, ScalarField, mul), Point)
+import Test.QuickCheck.Property (withMaxSuccess)
+
+specMultiScalarMultiplication
+    :: forall
+        (a :: Type)
+        (b :: Natural)
+        (c :: Natural)
+    .   ( EllipticCurve a
+        , ScalarField a ~ Zp b
+        , KnownNat b
+        , BaseField a ~ Zp c
+        , KnownNat c
+        )
+    => IO ()
+specMultiScalarMultiplication = hspec $ do
```

Comment:
I suggest refactoring this test: what we actually want to check is that `CoreFunction` instances for Haskell and Rust implementations agree with each other. Also, maybe randomize the number of points (e.g., let it be an arbitrary number from 0 to 100).

Reply:
Done